test/runtime/connectivity.go

```diff
@@ -280,135 +280,200 @@ var _ = Describe("RuntimeValidatedConntrackTest", func() {
 		vm.ContainerCreate(helpers.Client, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.client")
 		vm.ContainerCreate(helpers.Server, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.server")
+		vm.ContainerCreate(helpers.Httpd1, helpers.HttpdImage, helpers.CiliumDockerNetwork, "-l id.httpd")
+		vm.ContainerCreate(helpers.Httpd2, helpers.HttpdImage, helpers.CiliumDockerNetwork, "-l id.httpd_deny")
+		vm.ContainerCreate(curl1ContainerName, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.curl")
+		vm.ContainerCreate(curl2ContainerName, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.curl2")
+
 		vm.PolicyDelAll()
+
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath("ct-test-policy.json"), helpers.HelperTimeout)
+		Expect(err).Should(BeNil())
+
 	})
 
 	AfterEach(func() {
 		if CurrentGinkgoTestDescription().Failed {
 			vm.ReportFailed()
 		}
 
-		vm.ContainerRm(helpers.Server)
-		vm.ContainerRm(helpers.Client)
+		containersToRm := []string{helpers.Client, helpers.Server, helpers.Httpd1, helpers.Httpd2, curl1ContainerName, curl2ContainerName}
+		for _, containerToRm := range containersToRm {
+			vm.ContainerRm(containerToRm)
+		}
 	})
-	It("Conntrack disabled", func() {
-		endpoints, err := vm.GetEndpointsIds()
-		Expect(err).Should(BeNil(), "could not get endpoints IDs")
 
-		status := vm.EndpointSetConfig(endpoints[helpers.Server], helpers.OptionConntrack, helpers.OptionDisabled)
-		Expect(status).Should(BeTrue(), fmt.Sprintf("could not set %s=%s on endpoint %q", helpers.OptionConntrack, helpers.OptionDisabled, helpers.Server))
-
-		status = vm.EndpointSetConfig(endpoints[helpers.Client], helpers.OptionConntrack, helpers.OptionDisabled)
-		Expect(status).Should(BeTrue(), fmt.Sprintf("could not set %s=%s on endpoint %q", helpers.OptionConntrack, helpers.OptionDisabled, helpers.Client))
+	It("Conntrack-related configuration options for endpoints", func() {
 
-		clientServerConnectivity()
-	})
-
-	It("ConntrackLocal disabled", func() {
+		By("ConntrackLocal configuration option disabled for each endpoint")
 		endpoints, err := vm.GetEndpointsIds()
 		Expect(err).Should(BeNil(), "could not get endpoint IDs")
 
-		status := vm.EndpointSetConfig(endpoints[helpers.Server], helpers.OptionConntrackLocal, helpers.OptionDisabled)
-		Expect(status).Should(BeTrue(), fmt.Sprintf("could not set %s=%s on endpoint %q", helpers.OptionConntrackLocal, helpers.OptionDisabled, helpers.Server))
+		conntrackOptionModes := []string{helpers.OptionDisabled, helpers.OptionEnabled}
+		for _, conntrackOptionMode := range conntrackOptionModes {
+			By(fmt.Sprintf("Testing ConntrackLocal endpoint configuration option: %s", conntrackOptionMode))
+			status := vm.EndpointSetConfig(endpoints[helpers.Server], helpers.OptionConntrackLocal, conntrackOptionMode)
+			Expect(status).Should(BeTrue(), fmt.Sprintf("could not set %s=%s on endpoint %q", helpers.OptionConntrackLocal, helpers.OptionDisabled, helpers.Server))
```

Comment:
Please test if the output of `cilium endpoint config <ID>` returns `ConntrackLocal Enabled`

Reply:
Added a function that sets an option and then checks it for the provided endpoint for all cases in this test.